src/main/webapp/js/instructorFeedbackEdit/visibilityOptions.js

```diff
@@ -39,29 +39,147 @@ function toggleVisibilityPreviewTab(clickedButton) {
     var $disabledInputs = $containingForm.find('input:disabled, select:disabled');
     $disabledInputs.prop('disabled', false);
 
-    updateEditTab($containingForm);
+    updateVisibilityCheckboxesDiv($containingForm);
 
-    updatePreviewTab($containingForm);
+    updateVisibilityMessageDiv($containingForm);
     $disabledInputs.prop('disabled', true);
 }
 
-function getVisibilityMessageIfPreviewIsActive(clickedButton) {
+function getVisibilityMessage(clickedButton) {
     var $containingForm = $(clickedButton).closest('form');
-    
-    if ($containingForm.find('.visibilityMessageButton').hasClass('active')) {
-        updatePreviewTab($containingForm);
-    }
+    updateVisibilityMessageDiv($containingForm);
+}
+
+/**
+ * bind events to the visibility dropdown menu to
+ *  - show/hide visibility checkboxes div
+ *  - update dropdown button text to reflected selected option
+ *  - update visibility message div
+ */
+function attachVisibilityDropdownEvent() {
+    $('body').on('click', '.visibility-options-dropdown-option', function(event) {
+        var $clickedElem = $(event.target);
+        var selectedOption = $clickedElem.data('optionName');
+        var $containingForm = $clickedElem.closest('form');
+
+        setVisibilityDropdownMenuText($clickedElem.html(), $containingForm);
+
+        var $editTab = $containingForm.find('.visibilityOptions');
+        if (selectedOption === 'OTHER') {
+            $editTab.show();
+            updateVisibilityCheckboxesDiv($containingForm);
+        } else {
+            // only uncheck all checkboxes and update accordingly if a common option is selected
+            uncheckAllVisibilityOptionCheckboxes($containingForm);
+            checkCorrespondingCheckboxes(selectedOption, $containingForm);
+            $editTab.hide();
+        }
+
+        updateVisibilityMessageDiv($containingForm);
+    });
+}
+
+/**
+ * bind click event of each visibility checkbox to update visibility message div
+ */
+function attachVisibilityCheckboxEvent() {
+    $('body').on('click', 'input[type="checkbox"][class*="Checkbox"]', function(event) {
+        var $containingForm = $(event.target).closest('form');
+        updateVisibilityMessageDiv($containingForm);
+    });
 }
 
 // ////////////// //
 // HELPER METHODS //
 // ////////////// //
 
+function setVisibilityDropdownMenuText(text, $containingForm) {
+    var $visibilityDropdown = $containingForm.find('.visibility-options-dropdown');
+
+    if (text === 'Custom visibility option...') {
+        $visibilityDropdown.find('button').html('Custom visibility option:');
```

Comment:
I'll change it to `Custom visibility options...` and `Custom visibility option:` because the second one is showing what custom option (singular) the user already selected